pkg/odo/cli/component/create.go

```diff
@@ -359,9 +359,16 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 
 		// Validate user specify registry
 		if co.devfileMetadata.devfileRegistry.Name != "" {
-			// TODO: We should add more validations here to validate registry existence and correctness
 			if co.devfileMetadata.devfilePath.value != "" {
-				return errors.New("You can specify registry via --registry if you want to use the devfile that is specified via --devfile")
+				return errors.New("You can't download devfile from registry by specifying --registry if you want to use the devfile that is specified via --devfile")
```

Comment:
Nit: I think the old error message was more clear, the new one is a little too long

Reply:
No problem, I can revert it back, I think the biggest problem for the old error message is it's missing "not". 

`You can specify registry` -> `You can't specify registry` because `--registry` and `--devfile` are mutual exclusive